chapter_2/exercise_2_02/loop.c

```diff
@@ -19,19 +19,11 @@ int main(void)
   {
     char c = getchar();
 
-    if (i >= (MAXLINE - 1))
+    if (i >= (MAXLINE - 1) || c == '\n' || c == EOF)
     {
       loop = 0;
+      break;
```

Comment:
@SamiNasry Can we have it without the `break` until we figure out what's up with that? I find it very educational to learn how to deal with the differences between different machines.

PS: I'm used to writing code in high-level programming languages, and I almost forgot about all these low-level implementation details.

Reply:
Yes I removed the break until we figure out why it acts like this. Thanks for your time.